smack-extensions/src/main/java/org/jivesoftware/smackx/pubsub/PubSubManager.java

```diff
@@ -289,16 +289,16 @@ public LeafNode getOrCreateLeafNode(final String id)
                     throw e2;
                 }
             }
+            if(e1.getXMPPError().getCondition() == Condition.service_unavailable) {
+                // This could be caused by Prosody bug #805 (see https://prosody.im/issues/issue/805). Prosody does not
+                // answer to disco#info requests on the node ID, which makes it undecidable if a node is a leaf or
+                // collection node.
+                LOGGER.warning("The PubSub service " + pubSubService
+                        + " threw an DiscoInfoNodeAssertionError, trying workaround for Prosody bug #805 (https://prosody.im/issues/issue/805)");
+                return getOrCreateLeafNodeProsodyWorkaround(id);
+            }
             throw e1;
         }
-        catch (PubSubAssertionError.DiscoInfoNodeAssertionError e) {
```

Comment:
I think this means that you can also delete the DiscoInfoNodeAssertionError and PubSubAssertionError class, no?